waltz-ng/client/data-flow/data-flow-edit.js

```diff
@@ -202,11 +202,20 @@ function controller($q,
     };
 
     vm.addSource = (app) => {
+        if(vm.app.id == app.id) {
+            notification.warning("An application may not link to itself.");
```

Comment:
should this be pulled out as it's common 
 
    If (notifyIllegalFlow(vm.app,app)) return;


Or similar ?